src/Output/MultipleFileOutput.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace Spatie\TypeScriptTransformer\Output;
+
+use Spatie\TypeScriptTransformer\TypeScriptTransformerConfig;
+
+class MultipleFileOutput extends Output
+{
+    public function __construct(protected TypeScriptTransformerConfig $config)
+    {
+        $path = $config->getOutputDestination();
+        $this->path = is_dir($path) ? rtrim($path, '/') : pathinfo($path, PATHINFO_DIRNAME);
+        $this->output = [];
+
+    }
+
+    public function append(string $typeScript, string $classPath = null): void
+    {
+        $pathName = ltrim(str_replace('\\', '/', $classPath), '/');
+
+        $this->output[$pathName] = $typeScript;
+    }
+
+    public function writeOut($fileType): array
+    {
+        $this->clearPrevious($fileType);
+
+        foreach ($this->output as $path => $contents) {
+            if ($contents === '') {
+                continue;
+            }
+
+            $path = "$this->path/$path.{$fileType}";
+
+            $this->ensureFilesExist($path);
+            file_put_contents(
+                $path,
+                $contents
+            );
+        }
+
+        return array_keys($this->output);
+    }
+
+    public function writesMultipleFiles(): bool
+    {
+        return true;
+    }
+
+    protected function clearPrevious(string $fileType): void
+    {
+        // In case there are previously defined types that are no longer needed
+        if (is_dir($this->path)) {
```

Comment:
Would it be usefull to use an external dependency like flysystem here? I'm not that big of a fan to add extra dependencies but this is code we probably don't want to maintain in the future.